File: treebolicWordNetServices/src/main/java/org/treebolic/wordnet/service/DownloadActivity.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.treebolic.wordnet.service;

import android.os.Bundle;
import android.widget.Toast;

import java.io.IOException;
import java.io.InputStream;

import androidx.annotation.NonNull;

/**
 * WordNet download activity
 *
 * @author Bernard Bou
 */
public class DownloadActivity extends org.treebolic.download.DownloadActivity
{
	/**
	 * Whether stream is tar.gz (zip otherwise)
	 */
	private boolean asTarGz;

	@Override
	protected void onCreate(final Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		this.downloadUrl = Settings.getStringPref(this, Settings.PREF_DOWNLOAD);
		if (this.downloadUrl == null || this.downloadUrl.isEmpty())
		{
			Toast.makeText(this, R.string.error_null_download_url, Toast.LENGTH_SHORT).show();
			finish();
		}
	}

	@Override
	public void start()
	{
		assert this.downloadUrl != null;
		this.asTarGz = this.downloadUrl.endsWith(".tar.gz");
		super.start(R.string.wordnet);
	}

	// P O S T P R O C E S S I N G

	@SuppressWarnings("SameReturnValue")
	@Override
	protected boolean doProcessing()
	{
		return true;
	}

	@SuppressWarnings("SameReturnValue")
	@Override
	protected boolean process(@NonNull final InputStream inputStream) throws IOException
	{
		new Deployer(DownloadActivity.this.getFilesDir()).process(inputStream, this.asTarGz);
		return true;
	}
}

File: treebolicWordNetServices/src/main/java/org/treebolic/wordnet/DownloadActivity.kt
/*
 * Copyright (c) Treebolic 2023. Bernard Bou <1313ou@gmail.com>
 */
package org.treebolic.wordnet

import android.os.Bundle
import android.widget.Toast
import java.io.IOException
import java.io.InputStream

/**
 * WordNet download activity
 *
 * @author Bernard Bou
 */
class DownloadActivity : org.treebolic.download.DownloadActivity() {

    /**
     * Whether stream is tar.gz (zip otherwise)
     */
    private var asTarGz = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        downloadUrl = Settings.getStringPref(this, Settings.PREF_DOWNLOAD)
        if (downloadUrl == null || downloadUrl!!.isEmpty()) {
            Toast.makeText(this, R.string.error_null_download_url, Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    public override fun start() {
        asTarGz = downloadUrl!!.endsWith(".tar.gz")
        super.start(R.string.wordnet)
    }

    override fun doProcessing(): Boolean {
        return true
    }

    @Throws(IOException::class)
    override fun process(inputStream: InputStream): Boolean {
        Deployer(filesDir).process(inputStream, asTarGz)
        return true
    }
}
